src/main/kotlin/com/hibob/academy/service/PetService.kt

```diff
@@ -18,4 +20,17 @@ class PetService(private val petDao: PetDao) {
         val owner = petDao.getPetOwner(petId)
         return owner ?: throw IllegalStateException("No owner with ID $petId")
     }
+
+    fun getOwnerPets(ownerId: Long): List<Pet> {
+        return petDao.getOwnerPets(ownerId)
+    }
+
+    fun countPetsByType(companyId: Long): Map<PetType, Int> {
```

Comment:
I'm using another func from petDao (getPetsByType) to adhere to DRY principle, so I didn't create a new func inside petDao for counting.
It seemed close enough for me.